gateway/webhook/webhook.go

```diff
@@ -261,6 +262,34 @@ func (webhook *HandleT) batchRequests(sourceDef string, requestQ chan *webhookT)
 	}
 }
 
+func prepareRequestBody(req *http.Request, includeQueryParams bool, sourceType string, sourceListForParsingParams []string) ([]byte, error) {
+	defer req.Body.Close()
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		return nil, errors.New(response.RequestBodyReadFailed)
+	}
+
+	jsonBody := []byte("{}") // Start with an empty JSON object if no body is present
+	if len(body) > 0 {
+		jsonBody = body
+	}
+
+	if includeQueryParams && slices.Contains(sourceListForParsingParams, strings.ToLower(sourceType)) {
+		queryParams := req.URL.Query()
+		paramsBytes, err := json.Marshal(queryParams)
+		if err != nil {
+			return nil, errors.New(response.ErrorInMarshal)
+		}
+
+		jsonBody, err = sjson.SetRawBytes(jsonBody, "query_parameters", paramsBytes)
+		if err != nil {
+			return nil, errors.New(response.InvalidJSON)
+		}
+	}
+	return jsonBody, nil
+}
```

Comment:
Thinking back, I say just updating the body in case if it is empty would be sufficient.
```suggestion
	if len(body) == 0 {
		body = []byte("{}") // If body is empty, set it to an empty JSON object
	}

	if includeQueryParams && slices.Contains(sourceListForParsingParams, strings.ToLower(sourceType)) {
		queryParams := req.URL.Query()
		paramsBytes, err := json.Marshal(queryParams)
		if err != nil {
			return nil, errors.New(response.ErrorInMarshal)
		}

		body, err = sjson.SetRawBytes(body, "query_parameters", paramsBytes)
		if err != nil {
			return nil, errors.New(response.InvalidJSON)
		}
	}
	return body, nil
}
```